ginS/gins.go

```diff
@@ -7,100 +7,97 @@ package ginS
 import (
 	"html/template"
 	"net/http"
-	"sync"
 
 	"github.com/gin-gonic/gin"
 )
 
-var once sync.Once
-var internalEngine *gin.Engine
+var engine *gin.Engine
 
-func engine() *gin.Engine {
-	once.Do(func() {
-		internalEngine = gin.Default()
-	})
-	return internalEngine
+func init() {
+	if engine == nil {
+		engine = gin.Default()
+	}
```

Comment:
How is this different from just writing:
```go
var engine = gin.Default()
```
? I don't think you can change the value of the `engine` variable before `init` is executed, so the if condition will always be true.